src/raft.c

```diff
@@ -177,6 +179,118 @@ void RaftExecuteCommandArray(RedisModuleCtx *ctx,
     }
 }
 
+unsigned int keyHashSlot(RedisModuleString * str);
+
+static void lockKeys(RedisRaftCtx *rr, raft_entry_t *entry)
+{
+    assert(entry->type == RAFT_LOGTYPE_LOCK_KEYS);
+
+    RaftReq *req = entry->user_data;
+
+    /* FIXME: can optimize this for leader by getting it out of the req, keeping code simple for now */
+    size_t num_keys;
+    RedisModuleString ** keys = RaftRedisLockKeysDeserialize(entry->data, entry->data_len, &num_keys);
```

Comment:
add ticket to optimize this to not have to deserialize from leader / node that has req.